src/bioregistry/parse_iri.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+"""Functionality for parsing IRIs."""
+
+from typing import Tuple, Union
+
+from bioregistry.resolve import get_format_urls, normalize_curie, parse_curie
+
+__all__ = [
+    'parse_iri',
+]
+
+_D = sorted(get_format_urls().items(), key=lambda kv: -len(kv[0]))
+
+
+def parse_iri(iri: str) -> Union[Tuple[str, str], Tuple[None, None]]:
+    """Parse a compact identifier from an IRI.
+
+    :param iri: A valid IRI
+    :return: A pair of prefix/identifier, if can be parsed
+
+    IRIs from Identifiers.org (https and http):
+
+    >>> parse_iri("https://identifiers.org/aop.relationships:5")
+    ('aop.relationships', '5')
+    >>> parse_iri("http://identifiers.org/aop.relationships:5")
+    ('aop.relationships', '5')
+
+    IRI from an OBO PURL:
+
+    >>> parse_iri("http://purl.obolibrary.org/obo/DRON_00023232")
+    ('dron', '00023232')
+
+    IRI from the OLS:
+
+    >>> parse_iri("https://www.ebi.ac.uk/ols/ontologies/ecao/terms?iri=http://purl.obolibrary.org/obo/ECAO_0107180")
+    ('ecao', '0107180')
+
+    IRI from native provider
+
+    >>> parse_iri("https://www.alzforum.org/mutations/1234")
+    ('alzforum.mutation', '1234')
+
+    .. todo:: IRI with weird embedding, like ones that end in .html
+    """
+    if iri.startswith('https://www.ebi.ac.uk/ols/ontologies/'):
+        return _parse_purl(iri.rsplit('=', 1)[1])
+    if iri.startswith('http://purl.obolibrary.org/obo/TTO_1058367'):
+        return _parse_purl(iri)
+    if iri.startswith('https://identifiers.org/'):
+        return parse_curie(iri[len('https://identifiers.org/'):])
+    if iri.startswith('http://identifiers.org/'):
+        return parse_curie(iri[len('http://identifiers.org/'):])
```

Comment:
Should we add handling for old-style identifiers.org URLs? For instance http://identifiers.org/uniprot/P15056? These are still widely used.

Reply:
I thought about this. Probably not a bad idea. I will update the parse curie function to allow for different delimiters, since obo purls also use underscores instead

Reply:
Implemented in 0f180ed